code/modules/holiday/halloween/jacqueen.dm

```diff
@@ -0,0 +1,442 @@
+			var/mob/living/L2 = pick(GLOB.player_list)
+			objective += " [L2.name]."
+			brainwash(C, objective)
+		if(2)
+			visible_message("<b>[src]</b> waves their arms around, <span class='spooky'>\"Off comes your head, a pumpkin taking it's stead!\"</span>")
+			C.reagents.add_reagent("pumpkinmutationtoxin", 5)
+		if(3)
+			visible_message("<b>[src]</b> waves their arms around, <span class='spooky'>\"If only you had a better upbringing, your ears are now full of my singing!\"</span>")
+			var/client/C2 = C.client
+			C2.chatOutput.sendMusic("https://a.uguu.se/rQ8FxxUQ1Xzc_SpOwOkyOwOkyPumpkinSong-PFrPrIxluWk.mp4", 1)//I hope this works!
+		if(4)
+			visible_message("<b>[src]</b> waves their arms around, <span class='spooky'>\"You're cute little bumpkin, On your head is a pumpkin!\"</span>")
+			if(C.head)
+				var/obj/item/W = C.head
+				C.dropItemToGround(W, TRUE)
+			var/jaqc_latern = new /obj/item/clothing/head/hardhat/pumpkinhead/jaqc
+			C.equip_to_slot(jaqc_latern, SLOT_HEAD, 1, 1)
+		if(5)
+			visible_message("<b>[src]</b> waves their arms around, <span class='spooky'>\"In your body there's something amiss, you'll find it's a chem made by my sis!\"</span>")
+			C.reagents.add_reagent("eigenstate", 30)
+		if(6)
+			visible_message("<b>[src]</b> waves their arms around, <span class='spooky'>\"A new familiar for me, and you'll see it's thee!\"</span>")
+			C.reagents.add_reagent("secretcatchem", 30)
+		if(7)
+			visible_message("<b>[src]</b> waves their arms around, <span class='spooky'>\"While you may not be a ghost, for this sheet you'll always be it's host.\"</span>")
+			var/mob/living/carbon/human/H = C
+			if(H.wear_suit)
+				var/obj/item/W = H.wear_suit
+				H.dropItemToGround(W, TRUE)
+			var/ghost = new /obj/item/clothing/suit/ghost_sheet/sticky
+			H.equip_to_slot(ghost, SLOT_WEAR_SUIT, 1, 1)
+	poof()
+
+//Blame Fel
+/mob/living/simple_animal/jacq/proc/dating_start(mob/living/carbon/C, gender)
+	var/candies = pollGhostCandidates("Do you want to go on a date with [C] as Jacqueline the great pumpqueen?")
+	//sleep(30) //If the poll doesn't autopause.
+	if(candies)
+		candies = shuffle(candies)//Shake those ghosts up!
+		for(var/mob/dead/observer/C2 in candies)
+			if(C2.key && C2)
+				key = C2.key
+				message_admins("[C2]/[C2.key] has agreed to go on a date with [C] as Jacqueline.")
+				log_game("HALLOWEEN: [C2]/[C2.key] has agreed to go on a date with [C] as Jacqueline")
+				to_chat(src, "<span class='big spooky'>You are Jacqueline the great pumpqueen, witch Extraordinaire! You're a very Scottish lass with a kind heart, but also a little crazy. You also blew up the wizarding school and you're suspended for a while, so you visited the station before heading home. On your head lies the prize pumpkin of your Mother's pumpkin patch. You're currently on a date with [C] and well, I didn't think anyone would get this far. <i> Please be good so I can do events like this in the future. </i> </span>")
+				return
```

Comment:
You might want to, instead of just assigning a key, spawn a new `simple_animal` with the same sprite, and put some fanciness on it - For example, giving the controlling player the ability to end the date, releasing them back as a ghost and returning Jacq to her regular behavior.